File: PL/FinalProject/java/yu/pl/java/oop/DocumentProcessorBuilder.java
package yu.pl.java.oop;

import java.util.LinkedHashSet;
import java.util.LinkedHashMap;



public class DocumentProcessorBuilder  {

	private ConvertCase convertCase;
	private CountWords countWords;
	private FilterLines filterLines;
	private FindWords findWords;
	private NonAlphabeticFilter nonAlphabeticFilter;

	public void setConvertCase(ConvertCase convertCase) {
		this.convertCase = convertCase;
	}

	public void setCountWords(CountWords countWords) {
		this.countWords = countWords;
	}

	public void setFindWords(FindWords findWords) {
		this.findWords = findWords;
	}

	

	public void setFilterLines(FilterLines filterLines) {
		this.filterLines = filterLines;
	}

	public void setNonAlphabeticFilter(NonAlphabeticFilter nonAlphabeticFilter) {
		this.nonAlphabeticFilter = nonAlphabeticFilter;
	}

	public DocumentProcesor build() {
		return new DocumentProcesor(this);

	}

	public static class DocumentProcesor {
		LinkedHashSet<TextTask> tasks = new LinkedHashSet<TextTask>();
		private DocumentProcesor(DocumentProcessorBuilder builder) {
			/*this.convertCase = builder.convertCase;
			this.countWords = builder.countWords;
			this.filterLines = builder.filterLines;
			this.findWords = builder.findWords;
			this.nonAlphabeticFilter = builder.nonAlphabeticFilter;*/
			tasks = new LinkedHashSet<>();
			tasks.add(builder.filterLines);
			tasks.add(builder.convertCase);
			tasks.add(builder.findWords);
			tasks.add(builder.nonAlphabeticFilter);
			tasks.add(builder.countWords);
			
		}

		public LinkedHashMap process(String textFile) {

			Object temp = textFile;
			for(TextTask task: tasks) {
				if(task!= null) {
				temp = task.process(temp);
			}
			}
			if (!(temp instanceof LinkedHashMap)) {
				String str = new String((String)temp);
				 temp = new LinkedHashMap();
                      ((LinkedHashMap) temp).put(str,0);
			}
			return (LinkedHashMap) temp;


		}

	}

}

File: PL/FinalProject/OOPPython/DocumentProcessorBuilder.py
class DocumentProcessorBuilder(object):
	"""docstring for DocumentProcessorBuilder"""
	def __init__(self):
		self.Lower = None
		self.removeNonAlphabet = None
		self.FilterLines = None
		self.FindWords = None
		self.CountWords = None
		self.text = None;
	def setLower(self,lower):
		self.Lower = lower
	def setRemoveNonAlphabet(self,removeNonAlphabet):
		self.removeNonAlphabet = removeNonAlphabet
	def setFilterLines(self,FilterLines):
		self.FilterLines = FilterLines
	def setFindWords(self,FindWords):
		self.FindWords = FindWords
	def setCountWords(self,CountWords):
		self.CountWords = CountWords
	def setText(self, text):
		self.text = text

	def build(self):
		return DocumentProcessor(self)

class DocumentProcessor(object):
	def __init__(self, Builder):
		self.lower = Builder.Lower
		self.removeNonAlphabet = Builder.removeNonAlphabet
		self.FilterLines = Builder.FilterLines
		self.FindWords = Builder.FindWords
		self.CountWords = Builder.CountWords
		self.text = Builder.text
		self.builder = [ self.FilterLines, self.lower, self.FindWords, self.removeNonAlphabet, self.CountWords]


	def process(self):
		temp = self.text
		for i in self.builder:
			if i != None:
				temp = i.process(temp)

		return temp


